src/Raven.Server/Documents/Expiration/ExpirationStorage.cs

```diff
@@ -75,43 +75,54 @@ private void ThrowWrongExpirationDateFormat(Slice lowerId, string expirationDate
                 $"The expiration date format for document '{lowerId}' is not valid: '{expirationDate}'. Use the following format: {_database.Time.GetUtcNow():O}");
         }
 
-        public Dictionary<Slice, List<(Slice LowerId, string Id)>> GetExpiredDocuments(DocumentsOperationContext context,
-            DateTime currentTime, bool applyToExistingDocuments, long take, out Stopwatch duration, CancellationToken cancellationToken)
+        public class ExpiredDocumentsOptions
```

Comment:
Which parameters are required? All of them? If you do not have good ctors to guide future users or reasonable defaults, then in future it will result in someone sending a 0 in e.g. `AmountToTake`.